ray-operator/controllers/ray/common/pod.go

```diff
@@ -137,8 +137,6 @@ func DefaultHeadPodTemplate(instance rayiov1alpha1.RayCluster, headSpec rayiov1a
 	}
 	if dupIndex < 0 {
 		podTemplate.Spec.Containers[0].Ports = append(podTemplate.Spec.Containers[0].Ports, metricsPort)
-	} else {
```

Comment:
if i am reading this right, if user does not provide metrics port then we wont provide a default anymore.

Curious why we cannot have both?

1. By default exposes Prometheus on port 8080
2. Allow users to override

Reply:
Oh！ we will have the default Prometheus port 8080. 
```golang
	// add metrics port for exposing to the promethues stack.
	metricsPort := v1.ContainerPort{
		Name:          "metrics",
		ContainerPort: int32(DefaultMetricsPort),
	}
	dupIndex := -1
	for i, port := range podTemplate.Spec.Containers[0].Ports {
		if port.Name == metricsPort.Name {
			dupIndex = i
			break
		}
	}
        // dupIndex < 0 means the user does not set metricsPort, so we set the default metricsPort.
	if dupIndex < 0 {
		podTemplate.Spec.Containers[0].Ports = append(podTemplate.Spec.Containers[0].Ports, metricsPort)
	}
       // If the user does set metricsPort, we will now not override with the default metricsPort
```